tests/unit/test_dataclass_schema.py

```diff
@@ -40,3 +41,57 @@ def test_serialization_context():
 
     obj = MyObject.from_dict(dct)
     assert obj.sub_object.name == "testing"
+
+
+class MyEnum(StrEnum):
+    One = "one"
+    Two = "two"
+    Three = "three"
+
+
+@dataclass
+class SomeObject(dbtClassMixin):
+    name: str
+    an_attr: Optional[str] = None
+    an_int: int = 1
+    an_enum: Optional[MyEnum] = None
+    a_bool: bool = True
+
+
+def test_validation():
+    dct = {"name": "testing"}
+    SomeObject.validate(dct)
+    # check that use_default is not set in compile method
+    assert "an_attr" not in dct
+
+    dct = {"an_attr": "fubar"}
+    with pytest.raises(ValidationError) as excinfo:
+        SomeObject.validate(dct)
+    # former message: "'name' is a required property"
+    assert (
+        excinfo.value.message
+        == "Invalid value '{'an_attr': 'fubar'}': data must contain ['name'] properties"
```

Comment:
All the other error message look good - essentially saying X is invalid because of Y. 

In the case, the invalid value `'{'an_attr': 'fubar'}'` is invalid because it's _missing_ 1 or more properties. I think it's probably still helpful to name the invalid value (so users can identify where they need to go add the missing properties), so I feel good about this one as well.